guides/FRONTEND.md

```diff
@@ -0,0 +1,199 @@
+
+```graphql
+Map_by_pk(id: $id)
+```
+
+You can actually review all queries and mutations available in Hasura on `http://localhost:9695`.
+
+#### Integrating the GraphQL query
+
+In order to integrate the GraphQL query you need to use the `useQuery` function in the `urql` package. You will also need to import the GraphQL query created. The following showcases how to use the query with a React functional component.
+
+```typescript
+import React, { FC } from 'react';
+import { useQuery } from 'urql';
+import { GetMapQuery } from '/path/to/query';
+
+export const Component: FC = () => {
+  const id = 'uuid';
+
+  const [result] = useQuery({
+    query: GetMapQuery,
+    variables: { id },
+  });
+
+  const { data, fetching, error } = result;
+
+  return(
+    ...
+  )
+}
+```
+
+##### Optional, use useEffect
+
+You can also useEffect to rexecute queries based on changes to variables in GraphQL.
+
+```typescript
+import React, { FC, useEffect } from 'react';
+import { useQuery } from 'urql';
+import { GetMapQuery } from '/path/to/query';
+
+export const Component: FC = ({ id }) => {
+  const [result, reexecuteQuery] = useQuery({
+    query: GetMapQuery,
+    variables: { id },
+  });
+
+  useEffect(() => {
+    reexecuteQuery();
+    // Add variables to the array to trigger when the query is updated
+  }, [id]);
+
+  const { data, fetching, error } = result;
+
+  return(
+    ...
+  )
+}
+```
+
+Keep in mind the following parts of the query state:
+
+1. fetching is a boolean that indicates whether or not the query is loading
+
+2. error is whether or not an error was thrown executing the query
+
+3. data is the data returned from the query
+
+### Creating GraphQL mutations
+
+GraphQL mutations are used to update and insert new data into the database. The following is an example of inserting data into the `Map` table.
+
+```typescript
+import gql from 'fake-tag';
+
+export const CreateMap = gql`
+  mutation createMap($author: String!, $name: String!) {
+    insert_Map_one(
+      object: {
+        author_address: $author,
+        name: $name,
+        data: ""
+      }
+    ) {
+      id
+    }
+  }
+`;
+```
+
+For the most part, the key things to be mindful of syntax are the same as seen in the query section.
+
+#### Integrating the GraphQL mutation
+
+In order to use the mutation, you will need to use the `useMutation` function in the `urql` package. Mutations typically should only be done via user interactions. The following example showcases how a button triggers a mutation.
+
+```tsx
+import React, { FC } from 'react';
+import { useMutation } from 'urql';
+import { CreateMap } from '/path/to/query';
+
+export const Component: FC = () => {
+  const [state, executeMutation] = useMutation(CreateMap);
+  const { fetching, error, data } = state;
+
+  async function insertMap(author, name) {
+    await executeMutation({ author, name });
+  }
+
+  return(
+    <button type="button" onClick={async () => await insertMap('Hello World', 'Test 1234')}>
```

Comment:
Avoid putting asynchronous function in synchronous event handlers. Errors will be lost here. 

Better write something like 
```
function insertMap(author, name) {
   executeMutation({ author, name })
      .catch(console.error); // or stuff
  }
```

Reply:
That is true!
Technically, you can write a try/catch in the event handler.

But the example provided is wrong.
Thanks, will update.

Reply:
React event handlers are synchronous. Putting a try/catch could be a solution but the function would still be asynchronous. If by any mischance it throws, it leads to an unhandled exception, which is not good